src/dmd/mtype.d

```diff
@@ -3290,9 +3289,9 @@ extern (C++) final class TypeBasic : Type
         return (flags & TFlags.unsigned) != 0;
     }
 
-    override bool ischar() const
+    override bool isChar() const
```

Comment:
Move this to line 1046 and make it `final`. No need to use a virtual function.

Reply:
`virtual` makes the following possible: https://github.com/dlang/dmd/blob/b6cb784c7f5994cf15e37847ae622cd71dbf0b4a/src/dmd/mtype.d#L5820-L5823

Reply:
Add it when it's needed, not before.

Reply:
this is essentially a fix for what would be a silent false negative in the future i.e. `TypeEnum.isChar()` if used in the future will falsely return `false` without any compiler error or notice, possibly wasting debug time.

Reply:
Inventing abstractions without a clear use case, or any use case at all, generally doesn't work. At best, it creates dead code, complicates the code base, and leaves the user wasting his time wondering what the purpose of it is. In this example, the caller has already called toBasetype(), which looks through the enum. You propose using another virtual function to repeat it.

Besides, if it was really needed in isChar(), then isChar() could call toBasetype(). Another virtual function is not necessary.

Reply:
Fine.